src/sqlite.d

```diff
@@ -48,9 +49,16 @@ struct Database
 	{
 		// https://www.sqlite.org/c3ref/open.html
 		int rc = sqlite3_open(toStringz(filename), &pDb);
+		if (rc == SQLITE_CANTOPEN) {
+			// Database cannot be opened
+			writeln("\nThe database cannot be opened. Please check the permissions of ~/.config/onedrive/items.sqlite3\n");
```

Comment:
I'm wondering whether we should be logging the errors  rather than just spitting them out with writeln?  Also, I wonder whether a throw is more appropriate in these spots but encapsulating the method calls in try/catch perhaps?

Particularly as these can happen while the service is running and the user may not get any feedback otherwise?  But I could be wrong on that...

Reply:
I was thinking about all the logging for this change for a while. Certainly the above (cant open the database file) should go to the log file, however the others need to be potentially flagged as stderr.

Before submitting the patches I was working through trying to add try/catch however the application was still throwing the exception and exiting in a poor way.

Given the 'problem' being solved here I think writing to stderr is the right approach as we don't know what drive is full & writing to a log file when the disk is full would generate another exception.